src/Materials/PBR/babylon.pbrBaseMaterial.ts

```diff
@@ -487,13 +489,48 @@
             this._useLogarithmicDepth = value && this.getScene().getEngine().getCaps().fragmentDepthSupported;
         }
 
-        public needAlphaBlending(): boolean {
+        /**
+         * Gets the current transparency mode.
+         */
+        @serialize()
+        public get transparencyMode(): Nullable<number> {
+            return this._transparencyMode;
+        }
+
+        /**
+         * Sets the transparency mode of the material.
+         */
+        public set transparencyMode(value: Nullable<number>) {
+            if (this._transparencyMode === value) {
+                return;
+            }
+
+            this._transparencyMode = value;
+
+            this._forceAlphaTest = (value === PBRMaterial.PBRMATERIAL_ALPHATESTANDBLEND);
+
+            this._markAllSubMeshesAsTexturesDirty();
+        }
+
+        /**
+         * Specifies whether or not the meshes using this material should be rendered in alpha blend mode.
+         */
+        public needAlphaBlending(mesh: AbstractMesh): boolean {
             if (this._linkRefractionWithTransparency) {
                 return false;
             }
-            return (this.alpha < 1.0) || (this._opacityTexture != null) || this._shouldUseAlphaFromAlbedoTexture();
+
+            if (this._transparencyMode === PBRMaterial.PBRMATERIAL_OPAQUE ||

```

Comment:
what if I set alpha = 0.5? Do we want to give the last word to the transparencyMode?

Reply:
Yes, this is what the glTF spec says. https://github.com/KhronosGroup/glTF/tree/master/specification/2.0#alpha-coverage